File: src/main/java/com/damianryan/octopus/model/ElectricityMeterPoint.java
package com.damianryan.octopus.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AccessLevel;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

import java.util.List;

@Data
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ElectricityMeterPoint {

    String mpan;

    @JsonProperty("profile_class")
    int profileClass;

    @JsonProperty("consumption_standard")
    int consumptionStandard;

    List<ElectricityMeter> meters;

    List<Agreement> agreements;
}

File: src/main/kotlin/com/damianryan/octopus/model/ElectricityMeterPoint.kt
package com.damianryan.octopus.model

import com.fasterxml.jackson.annotation.JsonProperty

@JvmRecord data class ElectricityMeterPoint(
    @JsonProperty("gsp") val region: String? = null,
    val mpan: String,
    @JsonProperty("profile_class") val profileClass: Int = 0,
    @JsonProperty("consumption_standard") val consumptionStandard: Int? = 0,
    @JsonProperty("consumption_day") val consumptionDay: Int? = 0,
    @JsonProperty("consumption_night") val consumptionNight: Int? = 0,
    val meters: List<ElectricityMeter>,
    val agreements: List<Agreement>,
)
